test/integration/job/job_test.go

```diff
@@ -564,6 +578,46 @@ func TestJobPodFailurePolicy(t *testing.T) {
 		wantJobConditionType                     batchv1.JobConditionType
 		wantPodFailuresHandledByPolicyRuleMetric *metricLabelsWithValue
 	}{
+		"pod status matching the configured FailJob rule on exit codes when exit code=0; job terminated when JobPodFailurePolicy and AllowZeroInPodFailurePolicy enabled": {
```

Comment:
Let's park this integration test for 1.29 once we allow creation of jobs with 0 exit code.